www/perl-lib/SMT/Rest/SCCAPIv4.pm

```diff
@@ -359,6 +359,23 @@ sub delete_single_product
     my $guid = $self->user();
     my $c    = JSON::decode_json($self->read_post());
 
+    #
+    # lookup the Clients target
+    #
+    my $target = SMT::Utils::lookupTargetForClient($self->dbh(), $guid, $self->request());
+
+    #
+    # find Catalogs
+    #
+    my $existingregs = SMT::Utils::lookupRegistrationByGUID($self->dbh(), $guid, $self->request());
+    my @pidarr = keys %{$existingregs};
+    my $catalogs = SMT::Registration::findCatalogs($self->request(), $self->dbh(), $target, \@pidarr);
```

Comment:
It's impossible to deactivate the base product using single product deactivation (SUSEConnect checks that), so the only case when that check results in an error is when the client is not yet registered/is completely deregistered.